tests/e2e/events_test/get_filtered_events_test.go

```diff
@@ -552,14 +658,88 @@ func TestGetFilteredEventByPoints(t *testing.T) {
 	}
 }
 
+func TestGetFilteredPrivateEventsAsPublic(t *testing.T) {
+	CreateEvents()
+	defer ClearEvents()
+
+	received_events := models.EventList[models.EventDB]{}
+	api_err := errors.ApiError{}
+	response, err := public_client.New().Get("/event/filter/?isPrivate=true").Receive(&received_events, &api_err)
+	if err != nil {
+		t.Fatal("Unable to make request")
+		return
+	}
+	if response.StatusCode != http.StatusInternalServerError {
+		t.Fatalf("Request returned HTTP error %d", response.StatusCode)
+		return
+	}
+
+	expected_api_err := errors.ApiError{
+		Status:   http.StatusInternalServerError,
+		Type:     "DATABASE_ERROR",
+		Message:  "Could not fetch filtered list of events.",
+		RawError: "Invalid key isprivate",
+	}
+	if !reflect.DeepEqual(api_err, expected_api_err) {
+		t.Fatalf("Wrong event info. Expected %v, got %v", expected_api_err, api_err)
+	}
+}
+
+func TestGetFilteredPrivateEventsAsStaff(t *testing.T) {
+	CreateEvents()
+	defer ClearEvents()
+
+	received_events := models.EventList[models.EventDB]{}
+	response, err := staff_client.New().Get("/event/filter/?isPrivate=true").ReceiveSuccess(&received_events)
+	if err != nil {
+		t.Fatal("Unable to make request")
+		return
+	}
+	if response.StatusCode != http.StatusOK {
+		t.Fatalf("Request returned HTTP error %d", response.StatusCode)
+		return
+	}
+
+	expected_events := models.EventList[models.EventDB]{
+		Events: []models.EventDB{
+			{
+				EventPublic: models.EventPublic{
```

Comment:
I feel like this testevent1 creation can be extracted out into a function instead of being duplicated 50 times.

Reply:
This can be also addressed in the issue that I'd say I'd make.